### ISSUE TITLE
webcam detected but black on facetime

### ISSUE THREAD

**Issue by rforcen** [cobaohieu/ASUS-K55VD-HACKINTOSH#8]
appears to be correctly detected and recognized but displays black on FaceTime,

![screen shot 2018-04-29 at 13 55 05](https://user-images.githubusercontent.com/14297391/39406304-fbcbb03e-4bb4-11e8-82bf-fed526d7ce2e.png)

![screen shot 2018-04-29 at 13 55 42](https://user-images.githubusercontent.com/14297391/39406312-0c257bfe-4bb5-11e8-9504-cb642eb44755.png)

**Comment by cobaohieu** [comment]
Let's try this kext.
[CustomPeripheral.zip](https://github.com/southernvevo/ASUS-K55VD-HACKINTOSH/files/1998420/CustomPeripheral.kext.zip)

**Comment by evolutiondevel** [comment]
Use UVC Camera Control ( https://phoboslab.org/log/2009/07/uvc-camera-control-for-mac-os-x). Run skype , facetime, etc  and after  open UVC camera control.  Work for ASUS USB2.0 Webcam:  
ID product:	0x2883
ID vendor:	0x1bcf  (Sunplus Innovation Technology Inc.)
ID model:	UVC Camera VendorID_7119 ProductID_10371

**Comment by rforcen** [comment]
yes, works perfectly with provided instructions on mojave 10.14.3
thanks

**Comment by superman30** [comment]
Tenho o mesmo o problema no Notebook Asus VivoBook x512FJ, que possui a mesma web USB2.0 HD UVC WebCam: ID de Modelo:	UVC Camera VendorID_5075 ProductID_22237 ID Exclusivo:	0x1460000013d356dd. Fiz o procedimento no Catalina e continua dando tela preta. O que fazer? Tudo está funcionando menos a câmera. Poderia me dá uma luz? Desde já agradeço. Obrigado.

**Comment by superman30** [comment]
Does not work on Asus Notebook with the same cameraUSB2.0 HD UVC WebCam: Model ID: UVC Camera VendorID_5075 ProductID_22237 Unique ID: 0x1460000013d356dd
Could you give me a light? Thank you very much in advance!

**Comment by rforcen** [comment]
hi,

use the app. CameraControl to activate it:
https://phoboslab.org/log/2009/07/uvc-camera-control-for-mac-os-x
find link to a binary compressed file,
https://drive.google.com/file/d/1uPhogHz2xQzp0U0PEB-_U-jOnvdwKlBx/view?usp=sharing

best
bob

**Comment by superman30** [comment]
Hi, the camera control program incompatible with Catalina 10.15.4, I can't open it. Please help me. Then I make the donation, as I encourage your work. Thanks.

**Comment by cobaohieu** [comment]
Please try install all the kexts on folder method 1 and reboot. If kext load failed try the other method 2, install all kext on folder Clover->Kext->Other after that reboot ur PC. Please feedback if you had an issue.
[Method 2_CustomPeripheral.kext.zip](https://github.com/cobaohieu/ASUS-K55VD-HACKINTOSH/files/4398761/Method.2_CustomPeripheral.kext.zip)
[Method 1_AnyiSightCam.kext.zip](https://github.com/cobaohieu/ASUS-K55VD-HACKINTOSH/files/4398762/Method.1_AnyiSightCam.kext.zip)

**Comment by superman30** [comment]
Olá, amigo, os dois métodos infelizmente não funcionaram. Eles travaram na maça .... Mesmo assim, agradeço seu empenho, possui outra alternativa?

**Comment by cobaohieu** [comment]
> Olá, amigo, os dois métodos infelizmente não funcionaram. Eles travaram na maça .... Mesmo assim, agradeço seu empenho, possui outra alternativa?

Please take some shot about ur problems u had or ur issues.

**Comment by superman30** [comment]
Amigo, o fakeSMC e FakePcIID que travam o boot na maça. Sem os dois arquivos entro no sistema, mas a Camara UVC continua em tela preta. O que pode ser? O sistema reconhece a webcam, mas fica sem imagem. O que pode ser? Obrigado pela atenção

**Comment by cobaohieu** [comment]
> Amigo, o fakeSMC e FakePcIID que travam o boot na maça. Sem os dois arquivos entro no sistema, mas a Camara UVC continua em tela preta. O que pode ser? O sistema reconhece a webcam, mas fica sem imagem. O que pode ser? Obrigado pela atenção

Please upload ur folder Clover. So I could check for ur issues.

**Comment by superman30** [comment]
Ok! Segue, conforme solicitado

**Comment by rforcen** [comment]
can't update cameracontrol easily, it won't compile for catalina,
best
bob

**Comment by superman30** [comment]
Cobaohieu, poderia fazer esses ssdt e kext para UVC camera, conforme tutorial?

Make sure USB injection kexts or SSDT-UIA.aml are not loaded.
Download Hackintool by headkaze http://headsoft.com.au/download/mac/Hackintool.zip
Under the USB tab, identify the Bluetooth port and set it as internal. The UVC camera can also be set as internal. Export and obtain the codeless injection kext and SSDTs in ~/Desktop. Delete the SSDTs.
Install the USBPorts.kext in /L*/E* (Refer to https://www.tonymacx86.com/threads/guide-usb-power-property-injection-for-sierra-and-later.222266/ for more information.).
Rebuild cache and reboot.

**Comment by cobaohieu** [comment]
Please try install all the kexts on folder method 1 and reboot. If kext load failed try the other method 2, install all kext on folder Clover->Kext->Other after that reboot ur PC. Please feedback if you had an issue.

> [Method 2_CustomPeripheral.kext.zip](https://github.com/cobaohieu/ASUS-K55VD-HACKINTOSH/files/4398761/Method.2_CustomPeripheral.kext.zip)
[RehabMan-USBInjectAll-2018-1108.zip](https://github.com/cobaohieu/ASUS-K55VD-HACKINTOSH/files/4520973/RehabMan-USBInjectAll-2018-1108.zip)

> [Method 1_AnyiSightCam.kext.zip](https://github.com/cobaohieu/ASUS-K55VD-HACKINTOSH/files/4398762/Method.1_AnyiSightCam.kext.zip)
[RehabMan-USBInjectAll-2018-1108.zip](https://github.com/cobaohieu/ASUS-K55VD-HACKINTOSH/files/4520974/RehabMan-USBInjectAll-2018-1108.zip)

& Kext to patch for mac OS Catalina 10.15.X into CLOVER\config.plist\KernelAndKextPatches
Copy all text to config.plist, do not replace file config.plist.
[Kernel&Kextpatche.zip](https://github.com/cobaohieu/ASUS-K55VD-HACKINTOSH/files/4521003/Kernel.Kextpatche.zip)

**Comment by superman30** [comment]
Bom dia, amigo! Obrigado pelo seu tempo precioso, mas infelizmente tentei os dois métodos e não funcionaram. Desde já, agradeço seu tempo.

**Comment by superman30** [comment]
Amigo, infelizmente não funcionou para mim. Agradeço muito seu tempo. Obrigado.

**Comment by rforcen** [comment]
i still use mojave version

**Comment by JamesIslan** [comment]
> Bom dia, amigo! Obrigado pelo seu tempo precioso, mas infelizmente tentei os dois métodos e não funcionaram. Desde já, agradeço seu tempo.

Opa amigo, tenho um Asus X512FJ e estou tendo o mesmo problema com a USB 2.0 UVC Webcam (Azureware), você conseguiu resolver seu problema?

**Comment by VGerris** [comment]
I have the same issue with an HP HD camera. 
The link to the camera control app is nice, but I do not know how to recompile it, it is 32 bit so does not work on Montery.
The iSight kext does not work.

Are there any precise and specific instructions to follow? someone mentioned a Mojave description, I did not see the link?
Any help is greatly appreciated, thank you.

**Comment by elmaleek03** [comment]
> I have the same issue with an HP HD camera. The link to the camera control app is nice, but I do not know how to recompile it, it is 32 bit so does not work on Montery. The iSight kext does not work.
> 
> Are there any precise and specific instructions to follow? someone mentioned a Mojave description, I did not see the link? Any help is greatly appreciated, thank you.

hey man, i got my HP camera too and got the same issue. Have you solve ur problem?

**Comment by VGerris** [comment]
No I did not. Camera works in Linux. Haven't gotten any tool to display anything.

**Comment by cobaohieu** [comment]
Please check my repo to get the CameraControl.app on release or rebuild with Xcode
[https://github.com/cobaohieu/CameraControl-macOS-working-on-version-10.15-11.7/tree/main](https://github.com/cobaohieu/CameraControl-macOS-working-on-version-10.15-11.7/tree/main)

**Comment by VGerris** [comment]
hi, thanks for sharing, great to see a working version of it. I think I did either compile it or failed, or used another source selecter. The app works and the camera shows up as HP HD but it remains black, as with the other tools.
Maybe there is a trick to tell the cam to take the other source (hw id, there are two ) but last time I checked in depth I believe it was correct like in Linux and changing it didn't help ( but I might be wrong and good if someone can try too ).

I also tright some injection trick to make it look like an iSight without succes. I would love to see anyone fix this, there must be some secret service out there that would like for us to all have this working :) ?

**Comment by maverx72** [comment]
I want it too, but it still doesn't work. I connect the iPhone with a cable and use it as a camera.
